domains/client/domain-executor/src/system_bundle_processor.rs

```diff
@@ -218,14 +218,18 @@ where
             domain_block_result.header_number,
         );
 
+        self.domain_block_processor.on_domain_block_processed(
+            primary_hash,
+            domain_block_result,
+            head_receipt_number,
+        )?;
+
+        // TODO: The applied txs can be fully removed from the transaction pool
```

Comment:
There is a worker `txpool-notifications` spawn in the substrate `sc_service::spawn_tasks` doing this kind of work IIUC.